includes/html/pages/device/health.inc.php

```diff
@@ -16,252 +13,89 @@
-}
-
-if ($power) {
-    $datas[] = 'power';
-}
-
-if ($power_consumed) {
-    $datas[] = 'power_consumed';
-}
-
-if ($power_factor) {
-    $datas[] = 'power_factor';
-}
-
-if ($dBm) {
-    $datas[] = 'dbm';
-}
-
-if ($states) {
-    $datas[] = 'state';
-}
-
-if ($count) {
-    $datas[] = 'count';
-}
-
-if ($load) {
-    $datas[] = 'load';
-}
-
-if ($signal) {
-    $datas[] = 'signal';
-}
-
-if ($tv_signal) {
-    $datas[] = 'tv_signal';
-}
-
-if ($airflow) {
-    $datas[] = 'airflow';
-}
-
-if ($snr) {
-    $datas[] = 'snr';
-}
-
-if ($pressure) {
-    $datas[] = 'pressure';
-}
-
-if ($cooling) {
-    $datas[] = 'cooling';
-}
-
-if ($delay) {
-    $datas[] = 'delay';
-}
-
-if ($quality_factor) {
-    $datas[] = 'quality_factor';
-}
-
-if ($chromatic_dispersion) {
-    $datas[] = 'chromatic_dispersion';
-}
-
-if ($ber) {
-    $datas[] = 'ber';
-}
+unset($datas);
+$datas[] = 'overview';
+$type_text['overview'] = 'Overview';
 
-if ($eer) {
-    $datas[] = 'eer';
-}
+/*
+ * Main Sensors
+ */
 
-if ($waterflow) {
-    $datas[] = 'waterflow';
-}
+$main_sensors = [
+    'storage' => 'Storage',
+    'ucd_diskio' => 'Disk I/O',
+    'mempools' => 'Memory pools',
+    'processors' => 'Processors',
+];
 
-if ($loss) {
-    $datas[] = 'loss';
+foreach (array_keys($main_sensors) as $health) {
+    if (dbFetchCell('select count(*) from ' . $health . ' WHERE device_id = ?', [$device['device_id']])) {
+        $datas[] = $health;
+        $type_text[$health] = $main_sensors[$health];
+    }
 }
 
-if ($percent) {
-    $datas[] = 'percent';
-}
+    /*
+     * Sensors
+     */
 
-$type_text['overview'] = 'Overview';
-$type_text['charge'] = 'Battery Charge';
-$type_text['temperature'] = 'Temperature';
-$type_text['humidity'] = 'Humidity';
-$type_text['mempool'] = 'Memory';
-$type_text['storage'] = 'Disk Usage';
-$type_text['diskio'] = 'Disk I/O';
-$type_text['processor'] = 'Processor';
-$type_text['voltage'] = 'Voltage';
-$type_text['fanspeed'] = 'Fanspeed';
-$type_text['frequency'] = 'Frequency';
-$type_text['runtime'] = 'Runtime remaining';
-$type_text['current'] = 'Current';
-$type_text['power'] = 'Power';
-$type_text['power_consumed'] = 'Power Consumed';
-$type_text['power_factor'] = 'Power Factor';
-$type_text['dbm'] = 'dBm';
-$type_text['state'] = 'State';
-$type_text['count'] = 'Count';
-$type_text['load'] = 'Load';
-$type_text['signal'] = 'Signal';
-$type_text['tv_signal'] = 'TV signal';
-$type_text['airflow'] = 'Airflow';
-$type_text['snr'] = 'SNR';
-$type_text['pressure'] = 'Pressure';
-$type_text['cooling'] = 'Cooling';
-$type_text['delay'] = 'Delay';
-$type_text['quality_factor'] = 'Quality factor';
-$type_text['chromatic_dispersion'] = 'Chromatic Dispersion';
-$type_text['ber'] = 'Bit Error Rate';
-$type_text['eer'] = 'Energy Efficiency Ratio';
-$type_text['waterflow'] = 'Water Flow Rate';
-$type_text['loss'] = 'Loss';
-$type_text['qfp'] = 'QFP';
-$type_text['percent'] = 'Percent';
+    foreach (Sensor::getTypes() as $sensor) {
```

Comment:
This was kinda silly, should have been `DeviceCache::getPrimary()->sensors()->select('sensor_class')->selectSub('COUNT(*)', 'sensor_count')->groupBy('sensor_class')->pluck('sensor_count', 'sensor_class');`

```php
[
       "fanspeed" => 1,
       "power" => 1,
       "state" => 14,
       "temperature" => 3,
       "voltage" => 3,
     ]
```